### ISSUE TITLE
Add suppport for metdata explicit tag (itunes advisory)

### ISSUE THREAD

**Issue by limitlessbritt** [navidrome/navidrome#1357]
### Describe the solution you'd like

It would be nice to see the itunes advisory tag when I'm looking at songs in the web UI like how BPM was added.

![MusicBee_6BI5jKblCm](https://user-images.githubusercontent.com/37319328/134179422-fb150a25-503d-4120-acbb-1891002f260d.png)

**Comment by certuna** [comment]
I've been thinking about this before, it's not so difficult to add this. It's probably best to display it like it's shown in iTunes/Spotify, ie a little "E" icon next to the track name, rather than a whole extra column (which would take a lot of space).
As far as I know, the field is `TXXX:ITUNESADVISORY` in id3v2, `rtng` in MP4, but I don't know where this is commonly stored in FLAC/Vorbis - `EXPLICIT` or `CONTENTRATING`? `ITUNESADVISORY` sounds a bit weird as iTunes can't read FLAC.

**Comment by deluan** [comment]
Yeah, this should be simple to implement, and should be appended to the title as @certuna suggested. We can also append it server side, so it would be available to Subsonic clients.

**Comment by certuna** [comment]
would we want to append it to the title already serverside in the API endpoint, or do the "title + 🅴" appending client-side? that might interfere with other client-side concatenation logic like "title + subtitle"

**Comment by deluan** [comment]
> would we want to append it to the title already serverside in the API endpoint, or do the "title + 🅴" appending client-side? that might interfere with other client-side concatenation logic like "title + subtitle"

Yeah, that's why I think a `DisplayTitle` should be created on the server, with all this title/subtitle/movement/explicit logic, while also giving the client/ui the separated attributes.

**Comment by vikasazad** [comment]
Hi @deluan I wanted to work on this issue..

**Comment by metalheim** [comment]
> Yeah, that's why I think a `DisplayTitle` should be created on the server, with all this title/subtitle/movement/explicit logic, while also giving the client/ui the separated attributes.

Fully agree. Also Some Scrobblers could be put off if this is concatenated *in* the tracktitle.

**Comment by deluan** [comment]
@vikasazad, if you are still interested, go ahead!

@metalheim , hummm... I was thinking the other way around, but you are right, we have to display it as two separated elements so WebScrobbler and similar can still read the right title.

**Comment by certuna** [comment]
Still relevant and valuable, on my To Do list

**Comment by bruchar1** [comment]
What would be great is if a user could set an option in its preferences to prevent display and/or play of explicit tracks.

Also, if an admin user could prevent a given user to display / play explicit tracks.

**Comment by metalheim** [comment]
Not stale, would still like this feature

**Comment by certuna** [comment]
Good idea. Looks like there's consensus to add "Explicit" to the OpenSubsonic API, so when that's formalized we can implement it in Navidrome.

**Comment by certuna** [comment]
Bear in mind that explicit has been talked about but not standardized in the OpenSubsonic API - it can't be a simple boolean, since there's three possible states: none, clean and explicit.

**Comment by deluan** [comment]
This is now implemented _and_ standardized in OpenSubsonic API. Another one bites the dust!
